common/djangoapps/third_party_auth/saml.py

```diff
@@ -84,6 +85,17 @@ def generate_saml_config(self, idp=None):
         else:
             return super().generate_saml_config()
 
+    def auth_complete(self, *args, **kwargs):
+        """
+        Handle exceptions that happen during SAML authentication
+        """
+        try:
+            return super().auth_complete(*args, **kwargs)
+        # We are seeing errors of MultiValueDictKeyError looking for the parameter 'RelayState'.
+        # We would like to have a more specific error to handle for observability purposes.
+        except MultiValueDictKeyError as e:
+            raise AuthMissingParameter(self.name, e.args[0]) from e
```

Comment:
1. As noted in an outdated comment, can we add a unit test?
2. What is the difference between a real AuthMissingParameter and this type of AuthMissingParameter, and do we need to be able to differentiate at all? Maybe that's a question for @johnnagro/enterprise.

Reply:
1. Yeah, I can add a unit test, but I did want to make sure this was the strategy we wanted to take before I did any of that.
2. I could go looking for that, but the details are hidden in the django-social-auth code. I'm not sure why they don't handle this as a 'missing parameter' error, but the [saml backend just tries to get this value](https://github.com/python-social-auth/social-core/blob/master/social_core/backends/saml.py#L305) instead of [validating the state like the oauth backend does](https://github.com/python-social-auth/social-core/blob/master/social_core/backends/oauth.py#L194).

Reply:
Got it. You should definitely talk with Enterprise before moving forward on unit testing, etc. Feel free to include me on any Slack conversation. Thanks.

Reply:
LGTM. A test is super important here - I recently fixed a similar code path which lacked a test and had been broken through a linting exercise a number of years ago without notice.